### DISCUSSION TITLE
webui for remote access with the same frontend code?

### DISCUSSION THREAD

**Discussion by laoshaw** [webui-dev/webui#311] [Q&A]
webui embeds civetweb into a c and c++ program and it uses websocket to talk to civetweb, can I reuse the exact html+js+css of the GUI for remote access? e.g. I want to see the exact GUI from a remote machine on its browser, can webui expose the same port to external access? that way I can run it on my local machine, or on my server without code changes.

**Comment by laoshaw** [comment]
webui_set_public can make it externally accessible, but, it only allows one browser session, why can't two browser tabs access it at the same time? do I have to do a private frontend and a public frontend here even though both want to be the exact same UI?

**Comment by AlbertShown** [comment]
Yes, only one session is allowed per window. WebUI is not a web-server solution. Even if it's using web technologies, it is still a GUI library, not a web-server library.

**Issue:**
WebUI is window-based, not client/session-based, so, If you have a window that increments an _**n**_ number for every click, how can your backend distinguish the _**n**_ increment number for each user/session? When you run JavaScript from the backend, on which window it's supposed to run? or in all windows?

WebUI is a GUI, so think of it as windows, not sessions. Actually, WebUI adds a token security layer to make multiple sessions even harder for security reasons.

**Workaround:**
You can probably set the main window to never timeout, and on every connect, you create a new window and then redirect the user to this new window. I'm not sure if it will work but you can try.

**Reply by laoshaw** [reply]
got it, one session.
If I set the session public, I should be able to access it locally or remotely then, though at any given time, only one GUI can be open.
Thanks.
